src/txdb.cpp

```diff
@@ -307,12 +307,47 @@ bool CBlockTreeDB::WritePAKList(const std::vector<std::vector<unsigned char> >&
         return Write(std::make_pair(DB_PAK, uint256S("1")), offline_list) && Write(std::make_pair(DB_PAK, uint256S("2")), online_list) && Write(std::make_pair(DB_PAK, uint256S("3")), reject);
 }
 
-bool CBlockTreeDB::LoadBlockIndexGuts(const Consensus::Params& consensusParams, std::function<CBlockIndex*(const uint256&)> insertBlockIndex)
+bool CBlockTreeDB::WalkBlockIndexGutsForMaxHeight(int* nHeight) {
```

Comment:
This is not guaranteed to be the max height, correct? This is our best guess of the max height, from a random sample of 10k blocks?

Reply:
Could that be placed into the docstring for this method?

Reply:
Also, forgive my ignorance about the bitcoin / elements codebase, but is there no way to get the max height here? Certainly that is stored somewhere, since certain RPCs produce it.

Is it just that accessing it here would require some major refactors which are overkill for this PR?

Reply:
Sounds good. That makes sense, thanks,

What do you think about writing that "this is not guaranteed to be the max height, it's just from a random sample" in the method's doctoring?

Reply:
Added (hopefully I did the docstring right, there aren't a lot of them around.)